src/Microsoft.OData.Edm/Schema/EdmModel.cs

```diff
@@ -145,7 +145,27 @@ public void AddVocabularyAnnotation(IEdmVocabularyAnnotation annotation)
         public override IEnumerable<IEdmVocabularyAnnotation> FindDeclaredVocabularyAnnotations(IEdmVocabularyAnnotatable element)
         {
             List<IEdmVocabularyAnnotation> elementAnnotations;
-            return this.vocabularyAnnotationsDictionary.TryGetValue(element, out elementAnnotations) ? elementAnnotations : Enumerable.Empty<IEdmVocabularyAnnotation>();
+
+            this.vocabularyAnnotationsDictionary.TryGetValue(element, out elementAnnotations);
+
+            if (elementAnnotations == null)
+            {
+                elementAnnotations = Enumerable.Empty<IEdmVocabularyAnnotation>().ToList();
+            }
+
+            foreach (var kvp in this.vocabularyAnnotationsDictionary)
+            {
+                if (kvp.Key is IEdmAnnotationsTarget annotationsTarget)
+                {
+                    if (annotationsTarget.TargetSegments.Last() == element)

```

Comment:
> if (annotationsTarget.TargetSegments.Last() == element) [](http://example.com/codeflow?start=19&length=56)

we can't use this predict to find the declared vocabulary annotations. If it's schema definition, yes, we can do like this. Otherwise if it's target annotation, we should compare the whole path string.

Reply:
Reverted the entire change. We don't need it since we don't use schema definitions in `EdmAnnotationsTarget`